tests/Agent/IntegrationTests/IntegrationTests/Logging/Log4netMetricsAndForwardingTests.cs

```diff
@@ -220,6 +283,51 @@ public void LoggingWorksInsideTransaction()
             }
         }
 
+        [Fact]
+        public void AsyncLoggingWorksInsideTransaction()
+        {
+            if (_forwardingEnabled)
+            {
+                var expectedLogLines = new Assertions.ExpectedLogLine[]
+                {
+                new Assertions.ExpectedLogLine { LogLevel = "DEBUG", LogMessage = AsyncInTransactionDebugMessage, HasTraceId = true, HasSpanId = true },
+                new Assertions.ExpectedLogLine { LogLevel = "INFO", LogMessage = AsyncInTransactionInfoMessage, HasTraceId = true, HasSpanId = true },
+                new Assertions.ExpectedLogLine { LogLevel = "WARN", LogMessage = AsyncInTransactionWarningMessage, HasTraceId = true, HasSpanId = true },
+                new Assertions.ExpectedLogLine { LogLevel = "ERROR", LogMessage = AsyncInTransactionErrorMessage, HasTraceId = true, HasSpanId = true },
+                new Assertions.ExpectedLogLine { LogLevel = "FATAL", LogMessage = AsyncInTransactionFatalMessage, HasTraceId = true, HasSpanId = true },
+                };
+
+                var logLines = _fixture.AgentLog.GetLogEventDataLogLines();
+
+                Assertions.LogLinesExist(expectedLogLines, logLines);
+
+                Assert.Equal(expectedLogLines.Length, logLines.Where(x => x.Message.StartsWith("AsyncInTransaction")).Count());
+            }
+        }
+
+        [Fact]
+        public void AsyncNoAwaitLoggingWorksInsideTransaction()
+        {
+            if (_forwardingEnabled)
+            {
+                // NOTE: since the log is not awaited, it shows up outside the transaction
+                var expectedLogLines = new Assertions.ExpectedLogLine[]
+                {
+                new Assertions.ExpectedLogLine { LogLevel = "DEBUG", LogMessage = AsyncNoAwaitInTransactionDebugMessage},
```

Comment:
What do you think about making the fact that `HasTraceId` and `HasSpanId` are expected to be false in this case explicit, just to clarify what we mean by "it shows up outside the transaction"?

Reply:
Yes, but I'm going a different direction since this behavior is intermittent based on if the async logging takes place before the calling function returns. I'm planning to refactor the ExpectedLogLine data structure/helpers to use a nullable bool, which will make all the other tests more explicit.